src/dir.cr

```diff
@@ -93,6 +96,24 @@ class Dir
     String.new(dir).tap { C.free(dir as Void*) }
   end
 
+  def self.chdir path
+    if C.chdir(path) != 0
+      raise Errno.new("Error while changing directory")
+    end
+  end
+
+  def self.chdir(path)
+    old = working_directory
+    chdir(path)
+    yield
```

Comment:
@r20 as others noted, when a method `yields` for Crystal is a different overload. You can also make the "this method yields" explicit in the signature by providing a block argument (and not using it), like this:

```crystal
def foo
end

def foo(&block)
  yield
end
```

This has [no effect whatsoever](https://github.com/manastech/crystal/blob/master/src/compiler/crystal/semantic/normalizer.cr#L317) on the semantic or codegen, but it can indeed make the code a bit more explicit.

On the other hand, the [API docs](http://crystal-lang.org/api/Enumerable.html) will list them as having a different signature so you can spot them more easily.